tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/handler/AbstractQueryRequestHandler.java

```diff
@@ -134,6 +141,11 @@ public ModelAndView handleQueryRequest(HttpServletRequest request, RequestMethod
 
 	}
 
+	protected abstract Explanation explainQuery(final Query query, final Explanation.Level explainLevel);
```

Comment:
These aren't backwards compatible. We might want to have a default implementation that throws an exception. Not sure. Need to look at how this class is defined with regards to our versioning practice.